### PR TITLE
Clan app

### PR DESCRIPTION
**Pull request by louvenarde**
Replacement for the clan app

### CODE REVIEW

**Comment by Rackover** [comment]
The app has been thoroughly tested by a few people (Resistance, Samuel, Brutus, PhilipJFry, Blackheart, and more) and all errors / issues that arose have been solved. It's working fully and has feature parity for the legacy clan app. 

Ready for deployment.

File: public/js/app/browse_clans.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      $(".next").addClass("disabled");
+      $(".last").addClass("disabled");
+    }
+    
+    
+    let filter = "";
+    
+    if (searchName){
+        const components = searchName.replace(/\[/gi, "").replace(/\]/gi, "").split(' ');
+        const splitClan = [components.shift(), components.join(' ')];
```
Comment (Brutus5000) [inline public/js/app/browse_clans.js:39]: It is not easy to read kind of input you expect here. Please add comments with examples.

File: .env.example [hunk 2]
```diff
@@ -19,3 +19,5 @@ LEAGUES_UPDATE_INTERVAL = 3600              # one hour
 LEADERBOARDS_UPDATE_INTERVAL = 60           # one minute
 RECENT_USERS_LIST_UPDATE_INTERVAL = 900     # 15 minutes
 CLIENT_RELEASE_FETCHING_INTERVAL = 900      # 15 minutes
+CLAN_LIST_UPDATE_INTERVAL = 60              # one minute
+PLAYER_COUNT_UPDATE_INTERVAL = 3            # 3 seconds
```
Comment (Brutus5000) [inline .env.example]: Please add newline at the end.

File: templates/views/clans/browse.pug [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                                img(src="/images/loading.gif")
+    .container
+
+
+block js
+  script(type='text/javascript').
+    const clanList = !{JSON.stringify(clanList)};
+    const lastPage = !{lastPage};
+    const apiURL = !{JSON.stringify(apiURL)};
+  script(src="/js/browse_clans.min.js")
```
Comment (Brutus5000) [inline templates/views/clans/browse.pug]: Newline at the end

File: routes/views/clans/post/create.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  let locals = res.locals;
+
+  locals.formData = req.body || {};
+
+  let overallRes = res;
+
+  // validate the input
+  req.checkBody('clan_tag', 'Please indicate the clan tag - No special characters and 3 characters maximum').notEmpty().isLength({ max: 3 });
+  req.checkBody('clan_description', 'Please add a description for your clan').notEmpty().isLength({ max: 1000 });
+  req.checkBody('clan_name', "Please indicate your clan's name").notEmpty().isLength({ max: 64 });
```
Comment (Brutus5000) [inline routes/views/clans/post/create.js]: Just checked the db, max clan name there is 40

**Review by Brutus5000 — CHANGES_REQUESTED** [review]
Just some minor stuff

**Pull request merged**